rs/canister_sandbox/src/replica_controller/sandboxed_execution_controller.rs

```diff
@@ -1739,6 +1743,12 @@ fn evict_sandbox_processes(
         Backend::Empty => false,
     });
 
+    let total_sandboxes_rss = total_sandboxes_rss(backends);
+
+    if active_count <= max_active_sandboxes && total_sandboxes_rss <= max_sandboxes_rss {
+        return;
+    }
```

Comment:
Is this really needed? Wouldn't the first check inside `evict` do essentially the same? It now feels like we're leaking some of the concerns of `evict` here.

Reply:
Exactly, but if we return here, it will skip the creation of `Candidates` vec. This i just follow up on scheduler based eviction since Alin pointed this in one tread.